homeassistant/components/nightscout/sensor.py

```diff
@@ -120,3 +164,61 @@ def _parse_icon(self) -> str:
     def extra_state_attributes(self):
         """Return the state attributes."""
         return self._attributes
+
+
+class Battery(CoordinatorEntity, SensorEntity):
+    """Battery sensor of Nightscout device."""
+
+    # SENSOR_ATTR = "battery_percentage_remaining"
+    _attr_device_class = DEVICE_CLASS_BATTERY
+    _attr_state_class = STATE_CLASS_MEASUREMENT
+    _attr_native_unit_of_measurement = PERCENTAGE
+    _attr_entity_category = ENTITY_CATEGORY_DIAGNOSTIC
+
+    def __init__(self, coordinator, api: NightscoutAPI, name, unique_id):
+        """Initialize the Nightscout sensor."""
+        super().__init__(coordinator)
+        self.api = api
+        self._unique_id = unique_id
+        self._name = name
+        self._device_name = name
+        self._icon = "mdi:battery-unknown"
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the sensor."""
+        return self._unique_id
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def available(self):
+        """Return if the sensor data are available."""
+        return hasattr(self.coordinator.data[self._name], "uploader")
+
+    @property
+    def native_value(self):
+        """Return the state of the device."""
+        return self.coordinator.data[self._name].uploader.battery
+
+    @property
+    def icon(self) -> str:
```

Comment:
Please remove the icon property. The icon is handled automatically already by the frontend.